test/cppsuite/src/main/database_operation.cpp

```diff
@@ -385,13 +385,15 @@ database_operation::update_operation(thread_worker *tc)
         /* Get the cursor associated with the collection. */
         scoped_cursor &cursor = cursors[coll.id];
 
-        /* Choose a random key to update. */
-        testutil_assert(coll.get_key_count() != 0);
-        auto key_id =
-          random_generator::instance().generate_integer<uint64_t>(0, coll.get_key_count() - 1);
-        auto key = tc->pad_string(std::to_string(key_id), tc->key_size);
+        /* Choose a random key to update using a random cursor. */
+        scoped_cursor rnd_cursor = tc->session.open_scoped_cursor(coll.name, "next_random=true");
```

Comment:
Should we cache those random cursors? See the `map` created for the normal ones. @luke-pearson, what do you think?